spyder/plugins/editor/widgets/editor.py

```diff
@@ -515,8 +515,19 @@ def __init__(self, parent, actions):
            text=_("Show in Finder")
         else:
            text= _("Show in external file explorer")
-        external_fileexp_action = create_action(self, text,
-                                triggered=self.show_in_external_file_explorer)
+        external_fileexp_action = create_action(
+            self, text,
+            triggered=self.show_in_external_file_explorer,
+            shortcut=CONF.get_shortcut(context="Editor",
+                                       name="show in external file explorer"),
+            context=Qt.WidgetShortcut)
+
+        if self.parent() is not None:
+            plugin = self.parent().plugin
+            plugin.register_shortcut(external_fileexp_action,
+                                     context="Editor",
+                                     name="show in external file explorer",
+                                     add_shortcut_to_tip=True)

```

Comment:
This is not the right way to add a new shortcut. You should register it instead in the `create_shortcuts` method below.

The the problems you mentioned will simply go away.

Reply:
I must have left that part in assuming it was still needed from my testing. I already actually had an entry in `create_shortcuts` in fact.

Reply:
Great!